src/Mustard/Extension/Geant4X/Geometry/HelicalBox.c++

```diff
@@ -111,15 +116,80 @@ HelicalBox::HelicalBox(std::string name,
         c(i, 1) = AuxillaryPoint(u[i], 1);
     }
 
+    // compute end position and normal
+    const auto Helix{
+        [&](const auto& u) -> G4Point3D {
+            const auto u1{u + phi0};
+            return {radius * std::cos(u1),
+                    radius * std::sin(u1),
+                    u1 * tanAR - zOffset};
+        }};
+    const auto EndFaceNormal{
+        [&](const auto& u) -> G4ThreeVector {
+            const auto u1{u + phi0};
+            return {-radius * std::sin(u1),
+                    radius * std::cos(u1),
+                    tanAR};
+        }};
+    fFrontEndPosition = Helix(0);
+    fFrontEndNormal = EndFaceNormal(0).unit();
+    fBackEndPosition = Helix(phiTotal);
+    fBackEndNormal = EndFaceNormal(phiTotal).unit();
+
+    auto CreateParallelPoints{[&](const Eigen::MatrixX<G4Point3D>& grid,
```

Comment:
This thing should not be here. The projection should be done in `MainPoint` (line 79), and the procedure of making the full box is the same as before, by `x` itself.